packages/shared/src/variants/baduk.ts

```diff
@@ -282,10 +291,13 @@ export class GridBaduk extends Baduk {
   }
 }
 
-export const badukVariant: Variant<BadukConfig> = {
+export const badukVariant: Variant<NewBadukConfig> = {
   gameClass: Baduk,
   description:
     "Traditional game of Baduk a.k.a. Go, Weiqi\n Surround stones to capture them\n Secure more territory + captures to win",
   defaultConfig: Baduk.defaultConfig,
   getPlayerColors: Baduk.getPlayerColors,
+  sanitizeConfig: Baduk.sanitizeConfig,
 };
+
+export const gridBadukVariant = badukVariant as Variant<NewGridBadukConfig>;
```

Comment:
I thought about whether we can improve this, and have come up with the following:
```
export const badukVariant: Variant<NewBadukConfig> & {defaultConfig: () => NewGridBadukConfig} = {
  gameClass: Baduk,
  description:
    "Traditional game of Baduk a.k.a. Go, Weiqi\n Surround stones to capture them\n Secure more territory + captures to win",
  defaultConfig: Baduk.defaultConfig,
  getPlayerColors: Baduk.getPlayerColors,
  sanitizeConfig: Baduk.sanitizeConfig,
};

export const {sanitizeConfig: _, ...gridBadukVariant} = badukVariant;
gridBadukVariant satisfies Variant<NewGridBadukConfig>;
```

However I'm not sure if this is really an improvement. On the plus side, I believe this would be more typesafe.

Reply:
Oh that's interesting! I am not so familiar with `satisfies`, but it definitely seems better than `as`. I do assume we need to sanitize grid variants.  Let me play around a bit and see if we can do both.

(though maybe the right move is to just purge old games...)